Tasks/AzureFileCopyV2/PreJobExecutionAzureFileCopy.ts

```diff
@@ -0,0 +1,54 @@
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import fs = require('fs')
+import armStorage = require('azure-arm-rest/azure-arm-storage');
+import msRestAzure = require('azure-arm-rest/azure-arm-common');
+import { AzureRMEndpoint } from 'azure-arm-rest/azure-arm-endpoint';
+import { AzureEndpoint, StorageAccount } from 'azure-arm-rest/azureModels';
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+    if (isNonEmpty(resourceUri)) {
+        resourceUri = resourceUri.toLowerCase();
+        return resourceUri.substring(resourceUri.indexOf("resourcegroups/") + "resourcegroups/".length, resourceUri.indexOf("/providers"));
+    }
+
+    return "";
+}
+
+async function run(): Promise<void> {
+    let tempDirectory: string = tl.getVariable('Agent.TempDirectory');
+    let fileName: string = Math.random().toString(36).replace('0.', '');
```

Comment:
why can't we use the build or release id iteself for unique file name?

Reply:
Because there can be multiple tasks in the same definition.